operations/system_management_health/Configure_Prometheus_Email_Alert_Notifications.md

```diff
@@ -1,105 +1,108 @@
+          - match:
+              alertname: PostgresqlReplicationLagSMA
+            receiver:  email-alert
+          - match:
+              alertname: PostgresqlReplicationLagServices
+            receiver:  email-alert
+          - match:
+              alertname: PostgresqlFollowerReplicationLagSMA
+            receiver:  email-alert
+          - match:
+              alertname: PostgresqlFollowerReplicationLagServices
+            receiver:  email-alert
+        receivers:
+        - name: "null"
+        - name: email-alert
+          email_configs:
+          - to: receiver-email@yourcompany.com
+            from: sender-email@gmail.com
+            # Your smtp server address
+            smarthost: smtp.gmail.com:587
+            auth_username: sender-email@gmail.com
+            auth_identity: sender-email@gmail.com
+            auth_password: xxxxxxxxxxxxxxxx
+        EOF
+        ```
+
+1. Replace the alert notification configuration based on the files created in the previous step.
 
     ```bash
-    cat << 'EOF' > /tmp/alertmanager-new.yaml
-    global:
-      resolve_timeout: 5m
-    route:
-      group_by:
-      - job
-      group_interval: 5m
-      group_wait: 30s
-      receiver: "null"
-      repeat_interval: 12h
-      routes:
-      - match:
-          alertname: Watchdog
-        receiver: "null"
-      - match:
-          alertname: PostgresqlReplicationLagSMA
-        receiver:  email-alert
-      - match:
-          alertname: PostgresqlReplicationLagServices
-        receiver:  email-alert
-      - match:
-          alertname: PostgresqlFollowerReplicationLagSMA
-        receiver:  email-alert
-      - match:
-          alertname: PostgresqlFollowerReplicationLagServices
-        receiver:  email-alert
-    receivers:
-    - name: "null"
-    - name: email-alert
-      email_configs:
-      - to: receiver-email@yourcompany.com
-        from: sender-email@gmail.com
-        # Your smtp server address
-        smarthost: smtp.gmail.com:587
-        auth_username: sender-email@gmail.com
-        auth_identity: sender-email@gmail.com
-        auth_password: xxxxxxxxxxxxxxxx
-    EOF
+    ncn# sed "s/ALERTMANAGER_CONFIG/$(cat /tmp/alertmanager-new.yaml \
+                | base64 -w0)/g" /tmp/alertmanager-secret.yaml \
+                | kubectl replace --force -f -
     ```
 
-3.  Replace the alert notification configuration based on the files created in the previous step.
+1. Validate the configuration changes.
 
     ```bash
-    ncn-w001# sed "s/ALERTMANAGER_CONFIG/$(cat /tmp/alertmanager-new.yaml \
-    | base64 -w0)/g" /tmp/alertmanager-secret.yaml \
-    | kubectl replace --force -f -
+    ncn# kubectl exec alertmanager-cray-sysmgmt-health-promet-alertmanager-0 \
+                -n sysmgmt-health -c alertmanager -- cat /etc/alertmanager/config/alertmanager.yaml
     ```
 
-4.  Validate the configuration changes.
+    Check the logs for any errors if the configuration does not look accurate.
```

Comment:
should this be an individual step?

Reply:
I didn't think it made sense to be a top level step in the procedure, since it was conditional on the output of the previous kubectl command. But it occurs to me that I can turn this into a couple of sub-steps of the top-level validation step.